### ISSUE TITLE
Save gamefile's name with many dots is truncated when it is saved

### ISSUE THREAD

**Issue by telk5093** [OpenRCT2/OpenRCT2#18199]
### Operating System

Windows 10, 64 bits

### OpenRCT2 build

OpenRCT2, v0.4.2-3-g47d0402 (47d0402 on develop) provided by GitHub

### Describe the issue

Saving the park with named many dots, such as ``Some park name v1.1``, then it truncates the last term after the last dot and hence saved as ``Some park name v1``.
I'm guessing the game confuses it is an extension.

### Area(s) with issue?

New Save Format (NSF)

### Steps to reproduce

1. Save park as ``Some park name v1.1``
2. Check that the park is saved as ``Some park name v1``

### Attachments

![1](https://user-images.githubusercontent.com/10726524/194065096-0c621ed2-2ee9-41b8-800d-5c8b48460709.gif)

**Comment by CookiePLMonster** [comment]
How does it behave when you use a system file browser? Is it same or does it work properly then?

**Comment by telk5093** [comment]
> How does it behave when you use a system file browser? Is it same or does it work properly then?

It is fine when I use system file browser.

**Comment by beam41** [comment]
The save logic just replace last dot without checking, thinking it was extension
It should be like system file browser -> if extension match last dot do nothing if not just append extension to the file.
Can I work on this ? Seem like no one has working on it.

**Comment by CookiePLMonster** [comment]
> It should be like system file browser -> if extension match last dot do nothing if not just append extension to the file.

Yeah, that sounds like the best way. Just be mindful of cases where multiple extensions are "valid" - I'm not sure if there are any cases like this in the codebase, though.

**Comment by beam41** [comment]
we only have `.park` for game save and `.td6` for track save I think

Btw, I found there are check for this bug in system file browser
https://github.com/OpenRCT2/OpenRCT2/blob/7e653fc2016ad3a4421a85e4fa4641e612357fc3/src/openrct2-ui/windows/LoadSave.cpp#L427-L434

I will try to mimic this check.

**Comment by CookiePLMonster** [comment]
I'm not sure the system file browser bug is still valid after #18250.

**Comment by beam41** [comment]
Well, from my test, system file browser append extension correctly on its own, this line might not be important anymore, but I'm only tested on my win11 pc.

**Comment by CookiePLMonster** [comment]
I think the save dialog also needs "All Files" removed - there is no point in letting the user save the park with a custom extension if it's to be rejected further down the line anyway.

**Comment by beam41** [comment]
@CookiePLMonster I'd like to create another pull request for removing "All Files" but I'm not sure this logic can be remove too
https://github.com/OpenRCT2/OpenRCT2/blob/7e653fc2016ad3a4421a85e4fa4641e612357fc3/src/openrct2-ui/windows/LoadSave.cpp#L427-L434
I know in windows it fine, but I don't have linux machine.

**Comment by CookiePLMonster** [comment]
It probably can, considering the fact comments specifically mention Windows. You might want to wait until someone from the dev team doublechecks though.

**Comment by CookiePLMonster** [comment]
Any update on this?

**Comment by beam41** [comment]
I have made a pull request. That should work with the current version but I'm busy now so you can take over.
